src/Testcontainers.LowkeyVault/LowkeyVaultBuilder.cs

```diff
@@ -0,0 +1,334 @@
+    public LowkeyVaultBuilder WithCustomSslCertificate(string keyStoreFilePath, string keyStorePassword, StoreType keyStoreType = StoreType.PKCS12)
+    {
+        return WithEnvironment(LowKeyVaultEnvVarKey, AddOrAppend($"--server.ssl.key-store={keyStoreFilePath} "
+                                                               + $"--server.ssl.key-store-type={keyStoreType} "
+                                                               + $"--server.ssl.key-store-password={keyStorePassword ?? string.Empty}"))
+            .WithResourceMapping(new FileInfo(keyStoreFilePath), new FileInfo("/config/cert.store"));
+    }
+
+    /// <summary>
+    /// Enable Or Disable Debug.
+    /// </summary>
+    /// <param name="debug">The flag to enable or disable debug.</param>
+    /// <returns>A configured instance of <see cref="LowkeyVaultBuilder" />.</returns>
+    public LowkeyVaultBuilder WithDebug(bool debug)
+    {
+        return WithEnvironment(LowKeyVaultEnvVarKey, AddOrAppend($"--LOWKEY_DEBUG_REQUEST_LOG={debug}"));
+    }
+
+    /// <summary>
+    /// Enable Or Disable Relaxed Ports.
+    /// </summary>
+    /// <param name="relaxedPorts">The flag to enable or disable relaxed ports.</param>
+    /// <returns>A configured instance of <see cref="LowkeyVaultBuilder" />.</returns>
+    public LowkeyVaultBuilder WithRelaxedPorts(bool relaxedPorts)
+    {
+        return WithEnvironment(LowKeyVaultEnvVarKey, AddOrAppend($"--LOWKEY_VAULT_RELAXED_PORTS={relaxedPorts}"));
+    }
+
+    /// <summary>
+    /// Sets The host used to replace host placeholder in import template.
+    /// </summary>
+    /// <param name="logicalHost">The logical host.</param>
+    /// <returns>A configured instance of <see cref="LowkeyVaultBuilder" />.</returns>
+    public LowkeyVaultBuilder WithLogicalHost(string logicalHost)
+    {
+        return WithEnvironment(LowKeyVaultEnvVarKey, AddOrAppend($"--LOWKEY_IMPORT_TEMPLATE_HOST={logicalHost}"));
+    }
+
+    /// <summary>
+    /// Sets The port used to replace host placeholder in import template.
+    /// </summary>
+    /// <param name="logicalPort">The logical port.</param>
+    /// <returns>A configured instance of <see cref="LowkeyVaultBuilder" />.</returns>
+    public LowkeyVaultBuilder WithLogicalPort(ushort logicalPort)
+    {
+        return WithEnvironment(LowKeyVaultEnvVarKey, AddOrAppend($"--LOWKEY_IMPORT_TEMPLATE_PORT={logicalPort}"));
+    }
+
+
+    /// <summary>
+    /// Sets Additional Arguments.
+    /// </summary>
+    /// <param name="debug">The additional arguments.</param>
+    /// <returns>A configured instance of <see cref="LowkeyVaultBuilder" />.</returns>
+    public LowkeyVaultBuilder WithAdditionalArguments(List<string> additionalArguments)
+    {
+        return Merge(DockerResourceConfiguration, new LowkeyVaultConfiguration(additionalArguments: additionalArguments))
+            .WithEnvironment(LowKeyVaultEnvVarKey, AddOrAppend(string.Join(" ", additionalArguments)));
+    }
```

Comment:
I see you've put a lot of effort into exposing and supporting many of the available configurations. However, we usually only expose the absolutely necessary configurations to support the main use cases for running the container, and not any "nice-to-have" configurations. Developers can always refer to the image documentation and add additional configurations, e.g. by using environment variables.

Maintaining these APIs across different versions becomes very difficult. I suggest including only the essential ones.

In addition, we expect tests for any publicly available API.

Reply:
you are right, I have removed the "nice-to-have" APIs leaving only the necessary ones needed for the container to function.
I intentionally left `WithAdditionalArguments` API there as it provides an easy way for consumers to pass additional configurations taking into context the nuances surrounding passing env variables to Lowkey Vault.